rust/psibase/src/table.rs

```diff
@@ -46,25 +47,24 @@ pub trait TableRecord: PackableOwned {
 
 pub trait Table<Record: TableRecord>: Sized {
     const TABLE_INDEX: u16;
-    const TABLE_SERVICE: AccountNumber;
     const SECONDARY_KEYS: u8;
 
-    fn new(db_id: DbId, prefix: Vec<u8>) -> Self;
+    fn with_prefix(db_id: DbId, prefix: Vec<u8>) -> Self;
     fn prefix(&self) -> &[u8];
     fn db_id(&self) -> DbId;
 
-    fn open() -> Self {
-        let prefix = (Self::TABLE_SERVICE, Self::TABLE_INDEX).to_key();
+    fn new() -> Self {
+        let prefix = (get_service(), Self::TABLE_INDEX).to_key();
         Self::create_table_from_prefix(prefix)
     }
 
-    fn open_custom(service: AccountNumber, table_index: u16) -> Self {
+    fn with_service(service: AccountNumber, table_index: u16) -> Self {
         let prefix = (service, table_index).to_key();
         Self::create_table_from_prefix(prefix)
     }
 
     fn create_table_from_prefix(prefix: Vec<u8>) -> Self {
```

Comment:
Users providing the prefix is a rare case, so having just the 2-argument version is enough.

Reply:
the `create_table_from_prefix` is used from the `with_service()` and also from `new()`. I did not mean to expose this method but I guess we can't hide it in a trait.

Reply:
It's small enough to inline in the other functions